docs/decisions/app/0005-use-next-auth.md

```diff
@@ -0,0 +1,123 @@
+# Use NextAuth over custom/provider specific solution for authentication
+
+- Status: [proposed]
+- Deciders: @sawyerh, @aligg, @lorenyu
+- Date: 2023-07-20
+- Technical Story: As a user, I want to be able to easily integrate authentication into my react application with common identity providers, such as login.gov
+
+## Context and Problem Statement
+
+A common requirement for government projects is working with an identity provider (Such as login.gov or AWS Cognito). It isn't uncommon for authentication providers to be changed during the lifecycle of a project. We want to ease the concern of being "locked in" to one provider. [NextAuth](https://next-auth.js.org/) fills this need by allowing us to easily swap identity providers and providing helpful functions for retrieving user data.
+
+## Decision Drivers
+
+- **Extensible** - The user of this codebase can integrate with a variety of authentication providers.
+- **Maintained** - The NextAuth package is well-maintained and supported.
+- **Ease of use** - Developers can easily integrate the suggested approach to their applications.
+
+
+## Considered Options
+
+- NextAuth
+- Provider specific solution (such as [AWS Amplify](https://aws.amazon.com/amplify/) or [Azure AD](https://azure.microsoft.com/en-us/products/active-directory))
+- Custom authentication wrapper
+- Leave the responsibility to each project team
+
+## Decision Outcome
+
+In-progress
+
+## Pros and Cons of the Options
+
+### NextAuth
+
+Pros:
+
+1. **Well supported and maintained**: NextAuth is actively developed and has a strong community behind it, providing ongoing support and updates.
+2. **Extensible**: NextAuth supports a variety of built-in [Providers](https://next-auth.js.org/providers/), support for [custom Providers](https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider), and [Adapters](https://next-auth.js.org/adapters). Using a custom provider, it is easy to extend 
+3. Seamless integration with Next.js: NextAuth is designed to work seamlessly with Next.js applications. It provides a simple way to handle authentication in server-rendered and client-side rendered pages, ensuring a smooth user experience.
+
+Cons:
+1.	Dependency on Next.js: NextAuth is tightly coupled with Next.js framework. If you are not using Next.js, integrating NextAuth into your existing stack may require additional effort.
```

Comment:
This used to be true but they've recently pivoted to "Auth.js" and made it framework-agnostic: https://authjs.dev/ — for this template though, we'd still use the `next-auth` package.